### DISCUSSION TITLE
Migrate Startup Scripts to SQLite

### DISCUSSION THREAD

**Discussion by tanishqmanuja** [joshmedeski/sesh#55] [Ideas]
Since the project is in the early phase of development, it would be easier to switch now than later.

Using a file as a db could be unreliable and also not future proof.

**Unreliable as in:**
* If one extra character is inserted in toml the file will not parse and none of the workspace settings will work.

> [!NOTE]
> I am purposefully using the term workspace settings instead of just startup script here .. keeping in mind there could be more settings in future.

Also using SQLite is time tested for such things and used by many editors like vscode (.vscdb file) for example for the exact same use case.

**Consequently**, after migration we will have to expose a command to do basic operations:

```bash
sesh startup -add ./install.sh
sesh startup -remove
```
Both these commands implicitly use cwd as session path.

**Comment by joshmedeski** [comment]
We've already started playing with the idea of a sqlite databse:
https://github.com/joshmedeski/sesh/pull/42

My main concern is the added complexity to the user experience. Right now it's very simple to get started (just setup zoxide and the sessions will start coming in as you cd into projects). So there's a bit of work to be done before we have something stable and useful.

**Comment by tanishqmanuja** [comment]
Yeah, I have already seen that PR, but my discussion is very different than the PR. It's for ideas to replace the scripts in the config file. Currently we are using the config file as a DB.

**Comment by tanishqmanuja** [comment]
Also expanding on what you said on stream. We could have a table to sqlite to register some basic commands.

Maybe a workflow like:

**Register common scripts once**

```bash
sesh startup -register go ~/.config/sesh/bin/go.sh
```
Now the sqlite table looks like this

Name | Path
go | ~/.config/sesh/bin/go.sh

Now I sesh connect to a go project folder

**Adds the startup script to the current workspace**

```bash
sesh startup -add go
```
Now we check if `go` is an absolute path, which is not true so we can lookup into the existing sqlite registered command table and get the script path

Now we finally add this to the startup table in sqlite

Workspace | Script
/home/xyz/pathto/goproject | ~/.config/sesh/bin/go.sh

If this happens it would be a big PR for sure, maybe not suitable for now ... And as you have said, currently the test suite is important for sure 💯
